PowerShell/ScubaGear/Modules/Providers/ExportAADProvider.psm1

```diff
@@ -260,183 +240,200 @@ function Get-AADTenantDetail {
     }
 
-    # Process the Eligible role assignments if the premium license for PIM is there
-    if ($TenantHasPremiumLicense) {
-        # Get a list of all the users and groups that have Eligible assignments
-        $graphArgs = @{
-            "commandlet" = "Get-MgBetaRoleManagementDirectoryRoleEligibilityScheduleInstance"
-            "M365Environment" = $M365Environment }
-        $AllPIMRoleAssignments = Invoke-GraphDirectly @graphArgs
+    if ($Objecttype -eq "user") {
+        # If the user's data has not been fetched from graph, go get it and add it to the hashtable
+        if (-Not $PrivilegedUsers.ContainsKey($ObjectId)) {
+            $AADUser = Get-MgBetaUser -ErrorAction Stop -UserId $ObjectId
+            $PrivilegedUsers[$ObjectId] = @{"DisplayName"=$AADUser.DisplayName; "OnPremisesImmutableId"=$AADUser.OnPremisesImmutableId; "roles"=@()}
+Write-Warning "Processing role: $($RoleName) User: $($AADUser.DisplayName)"
+        }
+        # If the current role has not already been added to the user's roles array then add the role
+        if ($PrivilegedUsers[$ObjectId].roles -notcontains $RoleName) {
+            $PrivilegedUsers[$ObjectId].roles += $RoleName
+        }
+    }
 
-        # Add to the list of privileged users based on Eligible assignments
-        foreach ($Role in $AADRoles) {
-            $PrivRoleId = $Role.RoleTemplateId
-            # Get a list of all the users and groups Eligible assigned to this role
-            $PIMRoleAssignments = $AllPIMRoleAssignments | Where-Object { $_.RoleDefinitionId -eq $PrivRoleId }
-
-            foreach ($PIMRoleAssignment in $PIMRoleAssignments) {
-                $UserObjectId = $PIMRoleAssignment.PrincipalId
-                try {
-                    $UserType = "user"
-
-                    # If the user's data has not been fetched from graph, go get it
-                    if (-Not $PrivilegedUsers.ContainsKey($UserObjectId)) {
-                        $AADUser = Get-MgBetaUser -ErrorAction Stop -Filter "Id eq '$UserObjectId'"
-                        $PrivilegedUsers[$AADUser.Id] = @{"DisplayName"=$AADUser.DisplayName; "OnPremisesImmutableId"=$AADUser.OnPremisesImmutableId; "roles"=@()}
-                    }
-                    # If the current role has not already been added to the user's roles array then add the role
-                    if ($PrivilegedUsers[$UserObjectId].roles -notcontains $Role.DisplayName) {
-                        $PrivilegedUsers[$UserObjectId].roles += $Role.DisplayName
-                    }
+    elseif ($Objecttype -eq "group") {
+        # In this context $ObjectId is a group identifier so we need to iterate the group members
+        $GroupId = $ObjectId
+        # Get all of the group members that are transitively assigned to the current role via group membership
+        $GroupMembers = Get-MgBetaGroupMember -All -ErrorAction Stop -GroupId $GroupId
+Write-Warning "Processing role: $($RoleName) Group: $($GroupId)"
```

Comment:
not clear this is helpful, might be spammy.  make it write-debug?

Reply:
I created #1326 to handle a future update. We really need a way to trace the execution of the code because it is near impossible to troubleshoot in production, plus testing is hard without it too.